Cannot build rust nightly: unresolved import `crate::os::unix::net`

```
error[E0432]: unresolved import `crate::os::unix::net`
 --> library/std/src/os/net/linux_ext/addr.rs:3:22
  |
3 | use crate::os::unix::net::SocketAddr;
  |                      ^^^ could not find `net` in `unix`
```

```
/usr/ports/lang/rust-nightly/work/bootstrap/bin/rustc -vV
rustc 1.66.0-beta.1 (e080cc5a6 2022-11-01)
binary: rustc
commit-hash: e080cc5a659fb760c0bc561b722a790dad35b5e1
commit-date: 2022-11-01
host: x86_64-unknown-freebsd
release: 1.66.0-beta.1
LLVM version: 15.0.2
```

--- a/library/std/src/os/net/mod.rs
+++ b/library/std/src/os/net/mod.rs
@@ -1,4 +1,13 @@
 //! OS-specific networking functionality.
 
+// See cfg macros in `library/std/src/os/mod.rs` for why these platforms must
+// be special-cased during rustdoc generation.
+#[cfg(not(all(
+    doc,
+    any(
+        all(target_arch = "wasm32", not(target_os = "wasi")),
+        all(target_vendor = "fortanix", target_env = "sgx")
+    )
+)))]
 #[cfg(any(target_os = "linux", target_os = "android", doc))]
 pub(super) mod linux_ext;

